### DISCUSSION TITLE
Diff Tool Changed

### DISCUSSION THREAD

**Discussion by chris-zengenti** [VerifyTests/Verify#1010] [Q&A] [answered]
```
.NET 6.0
Rider 2023.1
Windows 11 Pro
```

Hi, 

I'm using Verify.NUnit latest version and Rider's built in Diff Tool.  However, today it has suddenly switched to Visual Studio.  I have DiffEngine_ToolOrder: Rider set in my global env variables but that doesn't seem to help.

How can I set it back?

Cheers
Chris

**Comment by SimonCropp** [comment]
i suspect you have updated Rider. and in a recent release they changed where the exe is located.
can you try updating to the latest version of Verify and see if that fixes it?

**Comment by chris-zengenti** [comment]
thanks for the response.  I'm using Verify.NUnit 21.3.0 (Verify 21.3.0)

**Reply by SimonCropp** [reply]
and it still doesnt work with 21.3.0?

If so what version of rider are you using?

**Comment by chris-zengenti** [comment]
sorry, correct - still doesn't work.  Here's my 'about' info:

```
JetBrains Rider 2023.1
Build #RD-231.8109.212, built on April 4, 2023
Runtime version: 17.0.6+10-b829.5 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
.NET 7.0.2 (Server GC)
GC: G1 Young Generation, G1 Old Generation
```

note, since my OP I have installed .NET7 to try and use the DiffEngine .net tool to see if that helped (it wouldn't let me with .NET6).  That also opens the diffs in Visual Studio.

**Reply by SimonCropp** [reply]
can u try rider 2023.2.2

**Comment by chris-zengenti** [comment]
didn't realised there was an available update, sorry.

i've tried that and get the same.  i.e. opens the diff in VS

**Reply by SimonCropp** [reply]
where is your rider exe located?

**Comment by chris-zengenti** [comment]
C:\Users\<username>\AppData\Local\JetBrains\Toolbox\apps\Rider\ch-0

I'm using the JetBrains tool box to install, probably should have mentioned to be honest.

**Comment by SimonCropp** [comment]
can u try v 22.1.3

**Reply by chris-zengenti** [reply]
can't see this version in my JetBrains ToolBox, there's a 22.3.3.

However, on a different project in the same Rider version I am finding that Rider is, correctly, using the Rider Diff Tool.

But on the other project, still using the VS one 🤷 .   I tried cloning the repo again thinking it might be some Rider user settings that had somehow got corrupted but I still get the same thing.  Colleagues using same repo and same Rider version don't get this issue btw.

**Reply by SimonCropp** [reply, marked as answer]
i meant v 22.1.3 of Verify

**Comment by chris-zengenti** [comment]
ok, thanks.  Yes, it's working now on 22.1.3 (and 22.1.4).  Thanks for your help.
